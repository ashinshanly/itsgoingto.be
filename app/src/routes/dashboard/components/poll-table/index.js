import PollTable from './poll-table'

export default PollTable
