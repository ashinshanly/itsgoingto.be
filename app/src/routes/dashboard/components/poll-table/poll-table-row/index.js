import PollTableRow from './poll-table-row'

export default PollTableRow
