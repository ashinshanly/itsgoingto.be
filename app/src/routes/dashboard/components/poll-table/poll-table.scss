@import "variables";

.panel {
  background: #fff;
  padding: 1.2em 1.6em;
  border: 1px solid #E6E9ED;
  border-radius: 2px;
  margin-bottom: 5em;

  .panel-header {
    padding-bottom: 0.6em;
    padding-left: 0.2em;
    font-size: 2em;
    line-height: 1em;
    font-weight: 300;
    border-bottom: 2px solid #e6e9ed;

    a {
      font-size: 0.8em;
      margin-left: 5px;
      cursor: pointer;
    }
  }

  .panel-body {
    position: relative;

    @media(max-width: $screen-xs-max) {
      overflow-x: auto;
    }

    .spinner-container {
      position: absolute;
      top: 2em;
      top: calc(50% - 2em);
      margin-left: auto;
      margin-right: auto;
      left: 0;
      right: 0;

      &.no-paginator {
        top: calc(50%);
      }
    }

    .table-container {
      opacity: 0.999;
      min-width: 665px;

      &.hidden {
        opacity: 0;
      }

      a {
        cursor: pointer;
      }
    }

    th {
      padding-left: 0.2em;
      padding-right: 0.2em;
      position: relative;

      a {
        color: $base-font-color;
      }

      i {
        position: absolute;
        right: 1.1em;
        line-height: 1.5em;
        color: $disabled-label-text-color;
      }

      &.sort-active:before {
        content : "";
        position: absolute;
        left: 0.2em;
        right: 1.1em;
        bottom  : 0;
        height  : 2px;
        border-bottom: solid 2px $base-font-color;
      }

      &.sort-active, &:hover {
        i {
          color: $base-font-color;
        }
      }
    }
  }
}
