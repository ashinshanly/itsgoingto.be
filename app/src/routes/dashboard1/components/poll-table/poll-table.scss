@import "variables";

.panel {
  background: #fff;
  padding:1.6em;
  border: 1px solid #E6E9ED;
  border-radius: 2px;
  margin-bottom: 5em;

  .panel-header {
    padding-bottom: 0.8em;
    padding-left: 0.6em;
    font-size: 2em;
    line-height: 1em;
    font-weight: 300;
    border-bottom: 2px solid #e6e9ed;

    a {
      font-size: 0.8em;
      margin-left: 5px;
      cursor: pointer;
    }
  }

  .panel-body {
    position: relative;

    @media(max-width: $screen-xs-max) {
      overflow-x: auto;
    }

    .spinner-container {
      position: absolute;
      top: 2em;
      top: calc(50% - 2em);
      margin-left: auto;
      margin-right: auto;
      left: 0;
      right: 0;

      &.no-paginator {
        top: calc(50%);
      }
    }

    .table-container {
      opacity: 0.999;
      min-width: 665px;

      &.hidden {
        opacity: 0;
      }

      a {
        cursor: pointer;
      }
    }

    th, td {
      text-align: center;
    }
  }
}
