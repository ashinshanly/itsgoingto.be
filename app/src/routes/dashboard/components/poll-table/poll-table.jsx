import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import CancelablePromise from 'cancelable-promise'
import { fetchPolls, deletePoll } from 'services/api'
import { pollsSelector, pollCountSelector, pollPageSelector } from 'store/poll/selectors'
import { POLLS_PER_PAGE } from 'store/poll'
import Spinner from 'components/spinner'
import PollTableRow from './poll-table-row'
import Paginator from '../paginator'
import './poll-table.scss'

let fetchPollsPromise

class PollTable extends React.Component {
  constructor (props) {
    super(props)

    const { pollPage, pollCount } = props

    // TODO : Store ID's of polls for the page in a dashboard store
    // Also store the page number in the dashboard store
    this.state = {
      page    : pollPage,
      loading : pollCount === 0
    }
  }

  componentDidMount = () => {
    const { page } = this.state

    this.fetchPollsForPage(page)
  }

  componentWillUnmount = () => {
    fetchPollsPromise.cancel()
  }

  fetchPollsForPage = (page) => {
    const { fetchPolls } = this.props

    fetchPollsPromise = new CancelablePromise(
      (resolve) => fetchPolls(page + 1).then(resolve)
    )
    fetchPollsPromise.then(() => {
      this.setState({ loading : false })
    })
  }

  changePage = (page) => {
    fetchPollsPromise.cancel()
    this.setState({ page, loading : true }, () => this.fetchPollsForPage(page))
  }

  render () {
    const { loading, page } = this.state
    const { polls, pollCount, deletePoll } = this.props
    const hasPaginator = Math.ceil(pollCount / POLLS_PER_PAGE) > 1

    return (
      <div className='panel'>
        <div className='panel-header'>
          Polls
        </div>
        <div className='panel-body'>
          {loading &&
            <div className={'spinner-container center-text' + (!hasPaginator ? ' no-paginator' : '')}>
              <Spinner />
            </div>
          }
          <div className={'table-container' + (loading ? ' hidden' : '')}>
            <table>
              <thead>
                <tr>
                  <th style={{ width: '4em' }}>ID</th>
                  <th style={{ width: '8em' }}>Identifier</th>
                  <th>Question</th>
                  <th style={{ width: '8em' }}>Responses</th>
                  <th style={{ width: '8em' }}>Status</th>
                  <th style={{ width: '12em' }}>Created At</th>
                  <th style={{ width: '4em' }}>Delete</th>
                </tr>
              </thead>
              <tbody>
                {
                  polls.map((poll) => (
                    <PollTableRow key={poll.id} poll={poll} deletePoll={deletePoll} />
                  ))
                }
              </tbody>
            </table>
          </div>
          <Paginator
            itemCount={pollCount}
            itemsPerPage={POLLS_PER_PAGE}
            page={page}
            pageCallback={this.changePage}
          />
        </div>
      </div>
    )
  }
}

PollTable.propTypes = {
  polls      : PropTypes.array.isRequired,
  pollCount  : PropTypes.number.isRequired,
  pollPage   : PropTypes.number.isRequired,
  fetchPolls : PropTypes.func.isRequired,
  deletePoll : PropTypes.func.isRequired,
}

const mapStateToProps = (state) => ({
  polls     : pollsSelector(state, true),
  pollCount : pollCountSelector(state),
  pollPage  : pollPageSelector(state)
})

const mapDispatchToProps = (dispatch) => ({
  fetchPolls : (page) => dispatch(fetchPolls(page)),
  deletePoll : (identifier) => dispatch(deletePoll(identifier))
})

export default connect(mapStateToProps, mapDispatchToProps)(PollTable)
