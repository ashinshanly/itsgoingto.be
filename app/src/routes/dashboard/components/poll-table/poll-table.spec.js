/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'
import { PollTable } from './poll-table'

const props = {
  polls       : [],
  pollCount   : 0,
  page        : 0,
  fetchPolls  : jest.fn(() => Promise.resolve()),
  setPollPage : jest.fn(),
  deletePoll  : jest.fn(),
}
let wrapper

describe('(Route) dashboard', () => {
  beforeEach(() => {
    wrapper = shallow(<PollTable {...props} />)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('(Component) poll table', () => {
    describe('(State)', () => {
      it('should have initial state for sorting', () => {
        expect(wrapper.state().sort).toBe('id')
        expect(wrapper.state().sortDirection).toBe('desc' )
      })
    })

    describe('(Action) onSort', () => {
      const testHeaderItem = (label, value) => {
        it('updates state to column ascending if not already sorted on column', () => {
          const headerItem = wrapper.find({ label })
          wrapper.setState({ sort: 'otherColumn' })
          headerItem.props().onSort()

          expect(wrapper.state().loading).toBe(true)
          expect(wrapper.state().sort).toBe(value)
          expect(wrapper.state().sortDirection).toBe('asc')
        })

        it('toggles state sortDirection if already sorted on column', () => {
          const headerItem = wrapper.find({ label })
          wrapper.setState({ sort: value, sortDirection: 'asc' })
          headerItem.props().onSort()

          expect(wrapper.state().loading).toBe(true)
          expect(wrapper.state().sort).toBe(value)
          expect(wrapper.state().sortDirection).toBe('desc')
        })

        it('should call fetch polls with sort and sortDirection', () => {
          const headerItem = wrapper.find({ label })
          wrapper.setState({ sort: value, sortDirection: 'asc' })
          headerItem.props().onSort()

          expect(props.fetchPolls).toHaveBeenCalledWith(1, value, 'desc')
        })
      }

      describe('ID', () => {
        testHeaderItem('ID', 'id')
      })

      describe('Identifier', () => {
        testHeaderItem('Identifier', 'identifier')
      })

      describe('Question', () => {
        testHeaderItem('Question', 'question')
      })

      describe('Responses', () => {
        testHeaderItem('Responses', 'responsesCount')
      })

      describe('Created At', () => {
        testHeaderItem('Created At', 'created')
      })
    })

    describe('(Render)', () => {
      describe('sort equals id asc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'id', sortDirection: 'asc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals id desc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'id', sortDirection: 'desc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals identifier asc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'identifier', sortDirection: 'asc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals identifier desc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'identifier', sortDirection: 'desc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals question asc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'question', sortDirection: 'asc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals question desc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'question', sortDirection: 'desc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals responsesCount asc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'responsesCount', sortDirection: 'asc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals responsesCount desc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'responsesCount', sortDirection: 'desc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals created asc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'created', sortDirection: 'asc' })

          expect(wrapper).toMatchSnapshot()
        })
      })

      describe('sort equals created desc', () => {
        it('should match snapshot', () => {
          wrapper.setState({ sort: 'created', sortDirection: 'desc' })

          expect(wrapper).toMatchSnapshot()
        })
      })
    })
  })
})
