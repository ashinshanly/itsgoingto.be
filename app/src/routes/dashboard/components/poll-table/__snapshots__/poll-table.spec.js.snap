// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) dashboard (Component) poll table (Render) sort equals created asc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection="asc"
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals created desc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection="desc"
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals id asc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection="asc"
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals id desc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection="desc"
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals identifier asc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection="asc"
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals identifier desc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection="desc"
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals question asc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection="asc"
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals question desc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection="desc"
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals responsesCount asc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection="asc"
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;

exports[`(Route) dashboard (Component) poll table (Render) sort equals responsesCount desc should match snapshot 1`] = `
<div
  className="panel"
>
  <div
    className="panel-header"
  >
    Polls
  </div>
  <div
    className="panel-body"
  >
    <div
      className="table-container"
    >
      <table>
        <thead>
          <tr>
            <Component
              label="ID"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
            <Component
              label="Identifier"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Question"
              onSort={[Function]}
              sortDirection={false}
              style={Object {}}
            />
            <Component
              label="Responses"
              onSort={[Function]}
              sortDirection="desc"
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Status"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "8em",
                }
              }
            />
            <Component
              label="Created At"
              onSort={[Function]}
              sortDirection={false}
              style={
                Object {
                  "width": "12em",
                }
              }
            />
            <Component
              label="Delete"
              onSort={null}
              sortDirection={null}
              style={
                Object {
                  "width": "4em",
                }
              }
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <Paginator
      itemCount={0}
      itemsPerPage={10}
      page={0}
      pageCallback={[Function]}
    />
  </div>
</div>
`;
